### ISSUE TITLE
Model with dropout in MirroredStrategy not work on latest keras and tf

### ISSUE THREAD

**Issue by jamm1985** [keras-team/keras#20611]
It is impossible to init model with dropout under any MirroredStrategy using tf API.

```python
import tensorflow as tf
STRATEGY = tf.distribute.MirroredStrategy()
with STRATEGY.scope():
    model = tf.keras.Sequential([
        tf.keras.layers.Dense(128, activation="gelu"),
        tf.keras.layers.Dropout(0.2),
        tf.keras.layers.Dense(128),
    ])
```

It seems that the problem is in the keras version: with `3.5.0` everything is fine, but `3.7.0` gives the error.

Related tf issue: https://github.com/tensorflow/tensorflow/issues/82281
